### PR TITLE
fix: Refactor annotations validation

### PR DESCRIPTION
**Pull request by lfvdavid**
Validation of annotated svcs.

Refactor validation function

### CODE REVIEW

**Comment by lfvdavid** [comment]
Tested on message-producer and this is the output fron the "print(errors)"

`[ValidationError(msg='Annotation refers to non-existing svc id: <sys-ed254:svc059_0>'), ValidationError(msg='Annotation refers to non-existing svc id: <sys-ed254:svc067_1, >'), ValidationError(msg='Annotation refers to non-existing svc id: <sys-ed254:svc50>'), ValidationError(msg='Annotation refers to non-existing svc id: <sys-ed254:svc51>')]`

The error is not otherwise printed when running the status command.

Atleast it finds and logs validation errors of annotations refering to non-existing svc id's

File: src/reqstool/common/validators/semantic_validator.py [hunk 1]
```diff
@@ -173,12 +173,8 @@ def _validate_annotation_tests_refers_to_existing_svc_ids(
...
         errors: List[ValidationError] = []
         for model, model_data in combined_raw_dataset.raw_datasets.items():
-            # Continue if model is not inital_urn
-            if (
-                model is not combined_raw_dataset.initial_model_urn
-                or not model_data.annotations_data
-                or not model_data.annotations_data.tests
-            ):
+            # Continue if no annotations data
+            if not model_data.annotations_data:
```
Comment (lfvbarcus) [inline src/reqstool/common/validators/semantic_validator.py:177]: shouldn't `or not model_data.annotations_data.tests` be added to the if statement also? If there are not annotated tests then there is nothing to compare the annotations towards?

Comment (lfvbarcus) [reply src/reqstool/common/validators/semantic_validator.py:177]: Not saying that it has to be there, it just seems appropriate?

Comment (lfvdavid) [reply src/reqstool/common/validators/semantic_validator.py:177]: Yes i could add it, tested now and the result is the same.
I guess it checks the same thing in a way, if there are no annotations there are no tests as the tests are collected from the annotations?

Comment (lfvbarcus) [reply src/reqstool/common/validators/semantic_validator.py:177]: Yes I get it, It's the discussion about what came first, the egg or the chicken. I'll approve.

**Review by lfvbarcus — APPROVED** [review]

**Pull request merged**